add: Address CRUD

truonghs opened the pull request:
# Issue

[[Issue task]](https://trello.com/c/n0qAtxRS)

## Summary

- Add apis to: Create, Update, Get, Delete, Set Default for Delivery address
### Media showcase

### Testing instructions:

## Reviews

### Author

- [x] I have reviewed my own code (all files marked as viewed in GitHub).
- [x] I have tested (via the postman) each of the interfaces/features related to the files changed/added.

truonghs committed:
add: Address CRUD

Devbeee commented on src/modules/address/address.controller.spec.ts:
```diff
@@ -0,0 +1,18 @@
+  it('should be defined', () => {
+    expect(controller).toBeDefined();
+  });
+});
```
ko viết test thì ko push

Devbeee commented on src/modules/address/address.controller.ts:
```diff
@@ -0,0 +1,115 @@
+    @Req() request: Request,
+  ) {
+    try {
+      const { id } = request['user'];
```
xài decorator @CurrentUser()

Devbeee commented on src/modules/address/address.controller.ts:
```diff
@@ -0,0 +1,115 @@
+export class AddressController {
+  constructor(private readonly addressService: AddressService) {}
+
+  @UseGuards(AuthGuard)
```
cai address này ng dùng phải xác thực ms dc thêm thì để lên đầu controller luôn ở dưới khỏi define lại

Devbeee commented on src/modules/address/address.controller.ts:
```diff
@@ -0,0 +1,115 @@
+  @ApiBadRequestResponse({ description: 'Missing input!' })
+  async getAddress(@Req() request: Request) {
+    try {
+      const { id } = request['user'];
```
xài decorator

Devbeee commented on src/modules/address/address.controller.ts:
```diff
@@ -0,0 +1,115 @@
+    @Req() request: Request,
+  ) {
+    try {
+      const { id } = request['user'];
```
xài decorator

Devbeee commented on src/modules/address/address.service.spec.ts:
```diff
@@ -0,0 +1,18 @@
+  it('should be defined', () => {
+    expect(service).toBeDefined();
+  });
+});
```
ko viết test thì ko push

Devbeee commented on src/modules/address/address.controller.ts:
```diff
@@ -0,0 +1,115 @@
+      return await this.addressService.getAddress(id);
+    } catch (error) {
+      throw error;
+    }
```
catch mà ko xử lý lỗi thì ko cần try catch

Devbeee commented on src/modules/address/address.controller.ts:
```diff
@@ -0,0 +1,115 @@
+    } catch (error) {
+      throw error;
+    }
+  }
```
catch mà ko xử lý lỗi thì ko cần try catch

Devbeee commented on src/modules/address/address.controller.ts:
```diff
@@ -0,0 +1,115 @@
+      return handleDataResponse('Address updated successfully!');
+    } catch (error) {
+      throw error;
+    }
```
catch mà ko xử lý lỗi thì ko cần try catch

Devbeee commented on src/modules/address/address.service.ts:
```diff
@@ -0,0 +1,135 @@
+        defaultAddressId: true,
+      },
+      where: {
+        id: Equal(userId),
```
id: userId dc r

Devbeee commented on src/modules/address/address.service.ts:
```diff
@@ -0,0 +1,135 @@
+        latitude: true,
+      },
+      where: {
+        userId: Equal(userId),
```
userId: userId

Devbeee commented on src/modules/address/address.service.ts:
```diff
@@ -0,0 +1,135 @@
+          defaultAddressId: true,
+        },
+        where: {
+          id: Equal(userId),
```
id: userId

Devbeee commented on src/modules/address/address.service.ts:
```diff
@@ -0,0 +1,135 @@
+            id: true,
+          },
+          where: {
+            userId: Equal(userId),
```
userId: userId

Devbeee commented on src/modules/address/dtos/delete-address.dto.ts:
```diff
@@ -0,0 +1,9 @@
+
+export class DeleteAddressDto {
+  @IsString({ message: 'AddressId must be a string' })
+  @IsNotEmpty({ message: 'Id is required' })
```
AddressId

Devbeee commented on src/modules/address/dtos/update-address.dto.ts:
```diff
@@ -0,0 +1,38 @@
+  @IsNotEmpty({ message: 'Address detail number is required' })
+  @ApiProperty()
+  addressDetail: string;
+}
```
Partial lại cái adddto là dc

Devbeee commented on src/modules/address/entities/address.entity.ts:
```diff
@@ -0,0 +1,74 @@
+  @ManyToOne(() => User, (user) => user.addresses)
+  @JoinColumn({ name: 'userId' })
+  owner: User;
+}
```
define 1 cai owner là dc r ko cần userId làm gì nữa,

Devbeee commented on src/modules/user/entities/user.entity.ts:
```diff
@@ -68,4 +76,13 @@ export class User {
   blogs: Blog[];
+
+  @ApiProperty()
+  @OneToMany(() => Address, (address) => address.userId)
```
address.owner

Devbeee commented on src/modules/user/user.controller.ts:
```diff
@@ -100,4 +104,40 @@ export class UsersController {
+  })
+  async getDefaultAddress(@Req() request: Request) {
+    try {
+      const { id } = request['user'];
```
xàii decorator

truonghs committed:
update: fix pull request

CRUD cart

fix pr

fix pr

fix: add to cart

fix pr

fix pull request

fix pull request

fix pull request

Devbeee commented on src/modules/address/address.service.ts:
```diff
@@ -0,0 +1,127 @@
+        latitude: true,
+      },
+      where: {
+        owner: Equal(user.id),
```
ko cần dùng Equal

Devbeee commented on src/modules/address/address.service.ts:
```diff
@@ -0,0 +1,127 @@
+          );
+        }
+      }
+      await this.addressRepository
+        .createQueryBuilder()
+        .delete()
+        .from(Address)
+        .where('id = :id', { id: deleteAddressDto.id })
+        .execute();
```
this.addressRepository.delete({ id: deleteAddressDto.id })

truonghs committed:
update: fix pull request

truonghs committed:
Merge branch 'dev' of github.com:Devbeee/BackEndFashionUITV2 into feature/address

truonghs committed:
Merge branch 'dev' of github.com:Devbeee/BackEndFashionUITV2 into feature/address
